enhancements/network/static-ip-addresses-vsphere.md

```diff
@@ -0,0 +1,613 @@
+---
+title: static-ip-addresses-vsphere
+authors:
+  - rvanderp3
+  - vr4manta
+reviewers: 
+  - JoelSpeed - machine API
+  - elmiko - machine API
+  - patrickdillon - installer
+  - jcpowermac - vSphere
+  - zaneb - networking
+  - 2uasimojo - hive
+approvers:
+  - JoelSpeed
+  - patrickdillon
+  - 2uasimojo
+api-approvers: 
+  - JoelSpeed
+creation-date: 2022-10-21
+last-updated: 2024-01-02
+tracking-link: 
+- https://issues.redhat.com/browse/OCPPLAN-9654
+see-also:
+  - /enhancements/installer/vsphere-ipi-zonal.md
+replaces:
+superseded-by:
+---
+
+# Static IP Addresses for vSphere IPI
+
+## Summary
+
+Static IP addresses are emerging as a common requirement in environments where
+the usage of DHCP violates corporate security guidelines.  Additionally, many 
+users which require static IPs also require the use of the IPI installer. 
+The proposal described in this enhancement discusses the implementation of
+assigning static IPs at both day 0 and day 2.
+
+## Motivation
+
+Users of OpenShift would like the ability to provision vSphere IPI clusters with static IPs.
+
+- https://issues.redhat.com/browse/OCPPLAN-9654
+
+### User Stories
+
+As an OpenShift administrator, I want to provision nodes with static IP addresses so that I can comply with my organization's security requirements.
+
+As an OpenShift administrator, I want to provision static IP addresses with the IPI installer so that I can reduce the complexity of certifying tools required to provision OpenShift.
+
+As an OpenShift administrator, I want to scale nodes with static IPs so that I can meet capacity demands as well as respond to disaster recovery scenarios.
+
+### Goals
+
+- All nodes created during the installation are configured with static IPs.  
+  Rationale: Many environments, due to security policies, do not allow DHCP.
+
+- The IPI installation method is able to provide static IPs to the nodes.
+  Rationale: Some users must qualify each tool used in their environment. 
+  Leveraging IPI greatly reduces the number of tools required to provision
+  a cluster.
+
+### Non-Goals
+
+- OpenShift will not be responsible for managing IP addresses.
+
+## Proposal
+
+### Static IPs Configured at Installation
+
+To facilitate the configuration of static IP address, network device configuration definitions are created for each node in the install-config.yaml. A `hosts` slice will be introduced to the installer platform specification to allow network device configurations to be specified for a nodes. 
+If a `hosts` slice is defined, enough `hosts` must be defined to cover the bootstrap, control plane, and compute nodes to be provisioned during the installation process.
```

Comment:
> Is it legal to do a hybrid install where e.g. control planes are static and workers are DHCP? The wording here implies not... but are we going to get asked for that?
yes, you could do this in a couple of ways that I can think of initially:
- create a compact cluster and create machinesets which use DHCP.  machinesets(or even replacement control plane nodes) dont have to use static IPs.
- perform a IPI install with all nodes having static IPs and create new machiensets/machines which dont use static IPs.

> Is it a requirement to peg each node to a specific role+IP? E.g. could I enumerate seven hosts entries with no role field and let installer distribute as it sees fit? And/or group everything in one entry...
we could have gone that route but the reason we didn't is that load balancers for the API and Ingress need to know which targets to expect. Sure, they could target the whole range of IPs but if they are installing with a large number of nodes this could become unwieldy imo.  

> Will the replicas fields be allowed at all?
Yes, but we expect the IPs for a given role to match the number of replicas.